ndscan/plots/image_2d.py

```diff
@@ -151,14 +151,17 @@ def _update(self):
         x_inds = _coords_to_indices(x_data[num_skip:num_to_show], self.x_range)
         y_inds = _coords_to_indices(y_data[num_skip:num_to_show], self.y_range)
 
-        z_min, z_max = self.current_z_limits
-        z_scaled = (z_data[num_skip:num_to_show] - z_min) / (z_max - z_min)
-
         cmap = colormaps.plasma
         if self._get_display_hints().get("coordinate_type", "") == "cyclic":
             cmap = colormaps.kovesi_c8
-        self.image_data[x_inds, y_inds, :] = cmap.map(z_scaled)
 
+        for x, y, z in zip(x_inds, y_inds, z_data):
+            self.image_data[x, y] = z
+
+        self.colorbar.setColorMap(cmap)
+        self.image_item.setColorMap(cmap)
```

Comment:
Presumably, this could be moved into `_initialise_series` now instead of re-setting the colour map every time?

Reply:
Yep, so I'll resolve this. Correct me if I've misunderstood your comment

Reply:
Actually, after checking - the `self.image_item.setColorMap(cmap` can be remove, it inherits from the `colorbar` anyway

Reply:
and the same applies for `self.image_item.setLevels`!